### DISCUSSION TITLE
"No assets found, exiting"

### DISCUSSION THREAD

**Discussion by enboig** [immich-app/immich#7560] [Q&A]
I am trying to upload to my server using docker-CLI, but it cannot find local files. I updated my command reading new documentation and upgraded local immich image:

```
$ docker pull ghcr.io/immich-app/immich-cli:latest
latest: Pulling from immich-app/immich-cli
Digest: sha256:eec411e35840da9e1b1b357d6ead1218777aa1b2951995b687ca84c0b96eecd5
Status: Image is up to date for ghcr.io/immich-app/immich-cli:latest
ghcr.io/immich-app/immich-cli:latest

$ docker run -it -v "$(pwd)":/import:ro -e IMMICH_INSTANCE_URL=http://192.168.1.X:2283/api -e IMMICH_API_KEY=xxxxx  -e IMMICH_RECURSIVE=true -e IMMICH_ALBUM_NAME="2015" ghcr.io/immich-app/immich-cli:latest upload
No assets found, exiting
```

Any hint?

**Comment by schuhbacca** [comment]
Are you running this command from a directory that has the photos in it you want? Do you need the recursive flag as well or is everything in the top level directory?

**Reply by enboig** [reply]
I am running in a folder which have images, and folders with images in them. I prefer using recursive flag, after reading a little should it be used like `-e IMMICH_RECURSIVE="/import"`? I tried but it is not working

**Reply by enboig** [reply]
```
# ls *.jpg | wc -l
5
# find . -iname "*.jpg" | wc -l
2988
```

**Reply by enboig** [reply]
`pwd` is a mouted md5 raid, but it shouldn't affect. Also file permission shouldn't be an issue, import folder is 775

**Comment by enboig** [comment]
I created a .env file to simplify command, but same result:
```
# cat ../.env 
IMMICH_INSTANCE_URL=http://192.168.1.x:2283/api
IMMICH_API_KEY=xxxxxxxxxx
IMMICH_RECURSIVE="/import"
IMMICH_ALBUM_NAME="2015"
IMMICH_INCLUDE_HIDDEN=true
IMMICH_DRY_RUN=true

# docker run -it -v "$(pwd)":/import --env-file="../.env"  ghcr.io/immich-app/immich-cli:latest upload
No assets found, exiting
```

**Reply by enboig** [reply]
Installing locally with npm worked

**Comment by kbknapp** [comment]
I've found this can happen if a directory or file has a single quote (`'`), such as `Family Pics '23/`. Renaming the directory or file to `Family Pics 2023/` fixes it.

**Reply by dcronin05** [reply]
This is the answer!

**Reply by himanshu0805** [reply]
just put "Folder path here" it worked for me

**Comment by kalven22** [comment]
The problem is in the command, try this. It works like charm!
docker run -it -v "$(pwd)":/import:ro -e IMMICH_INSTANCE_URL=https://YOUR_DOMAIN/api -e IMMICH_API_KEY=YOUR_API_KEY ghcr.io/immich-app/immich-cli:latest upload -a -c 5 --recursive /import

Be sure to **cd** into your photo directory before running this command.
